src/Parsers/ASTIdentifier.cpp

```diff
@@ -201,6 +201,39 @@ String ASTTableIdentifier::getDatabaseName() const
     else return {};
 }
 
+ASTPtr ASTTableIdentifier::getTable() const
+{
+    if (name_parts.size() == 2)
+    {
+        if (!name_parts[1].empty())
+            return std::make_shared<ASTIdentifier>(name_parts[1]);
+
+        if (name_parts[0].empty())
+            return std::make_shared<ASTIdentifier>("", children[1]->clone());
```

Comment:
should we also update name_parts in this case? because there are methods like `shortName()`, does it still work correctly?

Reply:
If `name_parts[0]` is empty then it means that there is a `ASTQueryParameter` in children vertor (cause it will not allow empty `ASTIdentifier` without a parameter that will later turn into an actual value), so I don't really see how exactly should we update `name_parts`.